pkg/metamorph/client.go

```diff
@@ -196,23 +196,38 @@ func (m *Metamorph) SubmitTransaction(ctx context.Context, tx *sdkTx.Transaction
 		}, nil
 	}
 
-	response, err := m.client.PutTransaction(ctx, request)
-	if err != nil {
-		m.logger.Warn("Failed to submit transaction", slog.String("hash", tx.TxID()), slog.String("key", err.Error()))
-		if m.mqClient != nil {
-			err := m.mqClient.PublishMarshal(SubmitTxTopic, request)
-			if err != nil {
-				return nil, err
+	var response *metamorph_api.TransactionStatus
+	var err error
+	// in case of error try PutTransaction until timeout expires
+	for {
+		start := time.Now()
+		response, err = m.client.PutTransaction(ctx, request)
+		duration := time.Since(start).Seconds() + 1
+		time.Sleep(300 * time.Millisecond)
+		if err != nil {
```

Comment:
The function could look a bit simpler if execution breaks in case of `err == nil`

I think this reflects essentially the same logic
```go
	start := m.now()
	const interval = 300 * time.Millisecond
	maxTimeout := time.Duration(request.MaxTimeout) * time.Second
	for {
		response, err = m.client.PutTransaction(ctx, request)
		if err == nil {
			break
		}
		m.logger.Warn("Failed to submit transaction", slog.String("hash", tx.TxID()), slog.String("key", err.Error()))

		time.Sleep(interval)

		if maxTimeout >= m.now().Sub(start) {
			continue
		}

		if m.mqClient != nil {
			err := m.mqClient.PublishMarshal(SubmitTxTopic, request)
			if err != nil {
				return nil, err
			}

			return &TransactionStatus{
				TxID:      tx.TxID(),
				Status:    metamorph_api.Status_QUEUED.String(),
				Timestamp: m.now().Unix(),
			}, nil
		}

		return nil, err
	}
```